theano/tensor/tests/test_opt.py

```diff
@@ -123,7 +125,7 @@ def test_merge2(self):
         dimshuffle_lift.optimize(g)
         self.assertTrue(str(g) == "[DimShuffle{0,1,x,x}(x)]", str(g))
         # Check stacktrace was copied over correctly after opt was applied
-        self.assertTrue(hasattr(g.outputs[0].tag, 'trace'))
+        self.assertTrue(check_stack_trace(g, ops_to_check='all'))
```

Comment:
If this check does not pass without your change ignoring Elemwise and DeepCopy, you should probably pass `ops_to_check=DimShuffle` here.